dev/workflow/hosts.py

```diff
@@ -24,57 +24,60 @@ def __init__(self, host=None):
             raise NotImplementedError(f'{host} is not a supported host.\n' +
                                       'Currently supported hosts are:\n' +
                                       f'{" | ".join(Host.SUPPORTED_HOSTS)}')
+        # If Host is instantiated with "host", use it
+        elif host is not None:
+            self.machine = host
+        # Otherwise, detect the host.
+        else:
+            # Detect the host if not provided
+            self.detect() if host is None else host
 
-        # Detect the host if not provided
-        detected_host = self.detect() if host is None else host
-
-        if host is not None and host != detected_host:
-            raise ValueError(
-                f'detected host: "{detected_host}" does not match provided host: "{host}"')
-
-        self.machine = detected_host
         self.info = self._get_info
         self.scheduler = self.info['SCHEDULER']
 
-    @classmethod
-    def detect(cls):
+    def __str__(self) -> str:
+        # The string representation of the Host object is the name of the machine
+        return self.machine
```

Comment:
```suggestion
        return f"{self.machine}"
```
or `return str(self.machine)`

This will ensure the `__str__` always returns a string representation of the class object.
This also allows you to replace `str(host)` with just `host` where it is being called.

Reply:
Updated.